docs/admin/authentication.md

```diff
@@ -580,6 +580,53 @@ Starting in 1.6, the ABAC and RBAC authorizers require explicit authorization of
 `system:anonymous` user or the `system:unauthenticated` group, so legacy policy rules
 that grant access to the `*` user or `*` group do not include anonymous users.
 
+## User impersonation
+
+A user can act as another user through impersonation headers, which let requests
+manually override the user info they authenticate as. For example an admin could
```

Comment:
it might be good to outline the flow:
1. An API request is authenticated, determining the requesting user, and setting them as the effective user.
2. If impersonation headers are present, an authorization check is performed to ensure the requesting user is allowed to impersonate every requested aspect of the impersonated user (username, groups, any extra attributes).
  a. If any of the impersonation authorization checks fails, the request is rejected with a forbidden error
  b. If all of the authorization checks succeed, the impersonation is logged, and the impersonated user becomes the effective user.
3. An authorization check is performed to ensure the effective user is allowed to perform the requested API operation. This means that in impersonation cases, only the impersonated user needs to be capable of performing the API operation.